File: kattis/cookingwater/cookingwater.py
n = int(input())
arr = []
for _ in range(n):
    a, b = list(map(int, input().split()))
    for i in range(a, b):
        arr.append((a, b))

for i in range(len(arr)):
    for j in range(i+1, len(arr)):
        if arr[i][1] < arr[j][0] or arr[i][0] > arr[j][1]:
            print("edward is right")
            exit(0)
print("gunilla has a point")

File: kattis/cookingwater/cookingwater.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n; cin >> n;
    vector< pair<int, int> > arr{};
    for (int i = 0; i < n; i++) {
        int a, b;
        cin >> a >> b;
        arr.push_back(make_pair(a, b));
    }
    for (int i = 0; i < arr.size(); i++) {
        for (int j = i+1; j < arr.size(); j++) {
            if (arr[i].second < arr[j].first || arr[i].first > arr[j].second) {
                cout << "edward is right" << endl;
                return 0;
            }
        }
    }
    cout << "gunilla has a point" << endl;
}